### ISSUE TITLE
Investigation: Can numpy improve parsing performance

### ISSUE THREAD

**Issue by cheahjs** [cheahjs/palworld-save-tools#156]
Investigate if using numpy to read and parse data structures could be faster than the current solution.

**Comment by magicbear** [comment]
no, you may check for the way that I used to load it by multiprocessing.
[MPMapPropertyProcess](https://github.com/magicbear/palworld-server-toolkit/blob/v0.8.1/palworld_server_toolkit/palobject.py#L724)
[MPMapProperty](https://github.com/magicbear/palworld-server-toolkit/blob/v0.8.1/palworld_server_toolkit/palobject.py#L522)

**Comment by magicbear** [comment]
24% performace from struct.Struct to int.from_bytes (like numpy)
>>> def perf():
...     t1 = time.time()
...     for _ in range(100000000):
...             struct.Struct("I").unpack(b"\x00\x00\x00\x00")
...     print(time.time()-t1)
...
>>>
>>> perf()
8.79693603515625
>>> def perf():
...     t1 = time.time()
...     for _ in range(100000000):
...             int.from_bytes(b"\x00\x00\x00\x01", byteorder="little")
...     print(time.time()-t1)
...
>>>
>>> perf()

**Comment by AntiMoron** [comment]
a suggestion: rewrite in rust. [[[[[
